### ISSUE TITLE
Xcode 8 using Swift 2.3

### ISSUE THREAD

**Issue by bensarz** [realm/realm-cocoa#4021]
## Goals

A version of Realm compiled with Xcode 8, using the `SWIFT_VERSION` setting set to 2.3 would be really awesome lifesaver for a lot of people. All of our dependencies are compilable using Xcode 8, all of them except for Realm.

I've compiled Realm using Xcode 8 and except for like 4 small changes, everything compiles fine.
It would be really appreciated if you could make a special release or at least provide the .zip file containing the Carthage structure so we could just drop it in until you release the Swift 3 version.

I've tried using your build script or packaging it using the carthage command, but that just send me down a Rabbit hole...

## Expected Results

1. make a release compilable with Xcode 8; or
2. make a zip containing the Carthage structure compiled with Xcode 8; or
3. ship Swift 3 right now?

**Comment by jpsim** [comment]
`master` supports the latest Xcode 8 beta (beta 6), with both Swift 2.3 and Swift 3. We can't make a release officially supporting Xcode 8 until Xcode 8 is out of beta.

**Comment by bensarz** [comment]
@jpsim I just tried using `master` and it built fine using Xcode 8.
However, the problem is that you cannot import a framework compiled using Swift 3.0 in another framework or app that is compiling for Swift 2.3.

I'm unsure if the opposite is possible though...

That said, I'll attempt to convert our codebase and get a sense of how much work is required to get everything working in Swift 3.0 and then we'll be able to start using the master branch.

Thanks for your prompt response @jpsim, as always!

**Comment by jpsim** [comment]
> However, the problem is that you cannot import a framework compiled using Swift 3.0 in another framework or app that is compiling for Swift 2.3.

You're right, it's not possible. But it's also not possible to specify more than one "default" Swift version to use in an Xcode project, so `Realm.xcodeproj` can't say "just use whatever Swift version the client project is using". So you'd need to modify that yourself (`SWIFT_VERSION=2.3`). It's just one thing to change.

> Thanks for your prompt response @jpsim, as always!

Happy to help.

**Comment by tpae** [comment]
Hello, I am on latest master and it's still failing for me..

/Pods/RealmSwift/RealmSwift/Object.swift:538:20: Use of unresolved identifier 'RLMDynamicGetByName'

:( Please help!

**Comment by jpsim** [comment]
You'll need to tell the RealmSwift pod to point to the same version of Realm if you want to use a custom version of Realm Swift:

```ruby
pod 'Realm', git: 'https://github.com/realm/realm-cocoa.git', branch: 'master', submodules: true
pod 'RealmSwift', git: 'https://github.com/realm/realm-cocoa.git', branch: 'master', submodules: true
```

**Comment by tpae** [comment]
@jpsim Thank you, it works! 👍

**Comment by alanshih** [comment]
I am using Swift 2.3 on xCode 8

Using 
```
pod 'Realm', git: 'https://github.com/realm/realm-cocoa.git', branch: 'master', submodules: true
pod 'RealmSwift', git: 'https://github.com/realm/realm-cocoa.git', branch: 'master', submodules: true
```

It is not working 
https://stackoverflow.com/questions/39663465/realm-using-xcode-8-swift-2-3-failing

**Comment by bensarz** [comment]
@alanshih if you follow [these instructions](https://realm.io/docs/swift/latest/#cocoapods-install) you should be able to use 2.3 without any problems.
Are there any reasons why you're using the `master` branch?

**Comment by alanshih** [comment]
@bensarz I was just following the instructions from @jpsim above. I deleted the Pods folder and then followed the instructions and did an install. I am still getting the same issue 72 missing files warning and 2 errors "Umbrella header 'Realm.h' not found" and "Could not build Objective-C module 'Realm'"

**Comment by bensarz** [comment]
@alanshih I opened this issue when Xcode 8 wasn't released officially and Realm hadn't released 1.1.0 yet. So I'd say, unless you _require_ using the `master` branch, I'd try to use the latest release instead.

**Comment by alanshih** [comment]
@bensarz  same issue. Followed the instructions you linked, but still having the same problem

**Comment by jpsim** [comment]
@alanshih please file a new issue with full details on your configuration and steps for us to reproduce and we'll gladly take a closer look.

**Comment by springhand** [comment]
help me, It is not working! 

Use:
Xcode 8
Swift 2.3

Podfile :
source 'https://github.com/CocoaPods/Specs'
platform :ios, '8.0'
use_frameworks!

target 'FOCSSwift' do

pod 'Alamofire' 
pod 'SwiftyJSON'
pod 'ZipArchive', '~> 1.4.0'
pod 'Realm', git: 'https://github.com/realm/realm-cocoa.git', branch: 'master', submodules: true
pod 'RealmSwift', git: 'https://github.com/realm/realm-cocoa.git', branch: 'master', submodules: true


end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    target.build_configurations.each do |config|
      config.build_settings['SWIFT_VERSION'] = '2.3' # or '3.0'
    end
  end
end


It is wrong,
Could not build Objective-C module 'Realm'
Umbrellaa header 'Realm.h' not found

**Comment by springhand** [comment]
O, Right
My cocoapods -v 1.1.1  Installation is 
Realm, 0.98.2
RealmSwift, 0.98.2

**Comment by jpsim** [comment]
@springhand 0.98.2 of the Realm and RealmSwift pods were released over 9 months ago, before Swift 2.3 was a thing. Please update your CocoaPods specs (`pod repo update`) to get the latest versions of our libraries (2.1.0 as of this writing).

In the future, please file new issues rather than commenting on months-old closed issues to make sure we don't miss your request!
